packages/create-wagmi/templates/next/src/wagmi.ts

```diff
@@ -1,11 +1,22 @@
 import { http, createConfig } from 'wagmi'
 import { mainnet, sepolia } from 'wagmi/chains'
-import { coinbaseWallet, injected, walletConnect } from 'wagmi/connectors'
+import {
+  coinbaseWallet,
+  injected,
+  walletConnect,
+  metaMask,
+} from 'wagmi/connectors'
 
 export const config = createConfig({
   chains: [mainnet, sepolia],
   connectors: [
     injected(),
+    metaMask({
+      dappMetadata: {
+        name: 'Create Wagmi',
+      },
+      infuraAPIKey: import.meta.env.VITE_INFURA_API_KEY,
```

Comment:
I don't think we should expose this at all as not all consumers use Infura, and tend to bring their own RPC providers. Instead, can we have the ability to pass an arbitrary RPC URL to the MetaMask SDK? This is what we are doing for the other connectors, like WalletConnect: https://github.com/wevm/wagmi/blob/main/packages/connectors/src/walletConnect.ts#L212-L217

This means we can infer the RPC URL from the provided `transports` from the Wagmi Config.

Reply:
Thanks, we have a similar way to configure rpcMap, will update accordingly.